src/components/Header.jsx

```diff
@@ -40,31 +41,55 @@ const Header = () => {
 
   const handleChange = (e) => {
     const { name, value } = e.target;
-
-    if (name === "confirmPassphrase") {
-      if (formValues.passphrase !== formValues.confirmPassphrase) {
-        setErrors({ passphrase: "Passphrases do not match" });
-      } else {
-        setErrors({ passphrase: "" });
+    setLoading(false);
+    setFormValues((prevValues) => {
+      const updatedValues = { ...prevValues, [name]: value };
+      if (name === "confirmPassphrase") {
+        if ((updatedValues.passphrase !== updatedValues.confirmPassphrase) && (updatedValues.passphrase && updatedValues.confirmPassphrase)) {
+          setErrors({ passphrase: "Passphrases do not match" });
+        } else {
+          setErrors({ passphrase: "" });
+        }
       }
-    }
-    setFormValues({ ...formValues, [name]: value });
+      return updatedValues;
+    });
   };
+
   const handleSubmit = (e) => {
     e.preventDefault();
-    mutate(formValues, {
-      onSuccess: (response) => {
-        setKeypair(response);
-        setStep(4);
-        toast.success(`Keypair Created successfully.`);
-      },
-      onError: (error) => {
-        setErrors(error.response.data);
-        for (const [attribute, errors] of Object.entries(error.response.data)) {
-          toast.error(errors[0]);
-        }
-      },
+    if (formValues?.passphrase && !formValues?.confirmPassphrase) {
+      toast.error("Please confirm your passphrase.");
+      return;
+    }
+
+    const filteredFormValues = { ...formValues };
```

Comment:
What if Rather than adding extra logic to handle empty values, we can simplify the handleChange function by deleting any key with an empty string value immediately. This way, we avoid rechecking for empty values later in the handle submit process process.

`const handleChange = (e) => {
    const { name, value } = e.target;
    setLoading(false);

    setFormValues((prevValues) => {
        // Create a new object by updating the specific form field value
        let updatedValues = { ...prevValues, [name]: value };

        // Remove the key if the value is an empty string
        if (value === "") {
            delete updatedValues[name];
        }

        // Handle passphrase validation (optional improvement based on updatedValues)
        const { passphrase, confirmPassphrase } = updatedValues;

        if (name === "confirmPassphrase" || name === "passphrase") {
            if (!passphrase || !confirmPassphrase) {
                setErrors({ passphrase: "" });  // Clear errors if either field is empty
            } else if (passphrase !== confirmPassphrase) {
                setErrors({ passphrase: "Passphrases do not match" });  // Set error if they don't match
            } else {
                setErrors({ passphrase: "" });  // Clear errors if they match
            }
        }

        return updatedValues;
    });
};`